<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>256</width>
      <height>421</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>функции</string>
     </property>
    </column>
    <item>
     <property name="text">
      <string>хэширование</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>контроль целостности</string>
     </property>
     <item>
      <property name="text">
       <string>формирование файла</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>проверка файла</string>
      </property>
     </item>
    </item>
    <item>
     <property name="text">
      <string>эцп</string>
     </property>
     <item>
      <property name="text">
       <string>выработка значения ЭЦП</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>проверка значения ЭЦП</string>
      </property>
     </item>
    </item>
    <item>
     <property name="text">
      <string>генерация ключевой пары</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_file_path">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>30</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вставьте путь к файлу</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="file_path">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>70</y>
      <width>591</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="file_path_complete">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>130</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>ОК</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_mac">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>30</y>
      <width>151</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вставьте путь к файлу</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_mac">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>70</y>
      <width>591</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_mac">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>130</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>ОК</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_hash">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>220</y>
      <width>591</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_mac">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>220</y>
      <width>601</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_mac_check">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>260</y>
      <width>351</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вставьте путь к файлу, который хотите проверить</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_mac_check">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>300</y>
      <width>601</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_mac_check">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>300</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_mac_check">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>420</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>ОК</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_hash_result">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>180</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Хэш-значение</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_hashfile_was_complete">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>110</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_mac_result">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>180</y>
      <width>281</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Значение файла контроля целостности</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_mac_file_complete">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>110</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_mac_file_for_check">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>340</y>
      <width>341</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вставьте путь к файлу контроля целостности</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_mac_for_check">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>370</y>
      <width>601</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_path_to_file_ecp">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>480</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вставьте путь к файлу</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_path_to_file_ecp">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>520</y>
      <width>601</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_key_for_ecp">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>580</y>
      <width>191</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите приватный ключ</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_key_for_ecp">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>620</y>
      <width>601</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_ok_ecp">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>670</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>ОК</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_ecp">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>540</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_ecp_result">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>720</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Значение ЭЦП</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_ecp_result">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>750</y>
      <width>601</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_key_gen">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>522</x>
      <y>20</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Хотите сгенерировать ключи?</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_key_gen">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>50</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>ДА</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_privkey">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>110</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Приватный ключ</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_privkey">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>140</y>
      <width>511</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_pubkey">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>260</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Публичный ключ</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_pubkey">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>290</y>
      <width>511</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_keys_ok">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>410</y>
      <width>291</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clear">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>680</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>clear</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1920</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
